pkg/cluster/internal/providers/docker/provision.go

```diff
@@ -255,6 +255,10 @@ func runArgsForNode(node *config.Node, clusterIPFamily config.ClusterIPFamily, n
 		args = append(args, "-e", "KUBECONFIG=/etc/kubernetes/admin.conf")
 	}
 
+	if len(node.Gpus) > 0 {
+		args = append(args, fmt.Sprintf("--gpus=%v", node.Gpus))
```

Comment:
let's not plumb this directly since the values are incompatible across backends

instead we can have a more structured format in the internal type (for now gpus true/false) and we need valdiatino on the external type.

Reply:
I added validation to ensure right now only "all" can be passed so it's now safe to to use node.Gpus directly in the docker provisioner. I prefer to keep the code simpler and passthrough instead of adding another layer of validation in the docker provisioning code. Happy to add additional validation in docker/povision.go if you have strong opinion on it.

For podman we can not plumb it through directly. So I will work out a way to convert node.Gpus into a podman compatible format. I don't have podman installed myself.